src/Consolonia.Core/Infrastructure/ArrowsAndKeyboardNavigationHandler.cs

```diff
@@ -15,9 +16,9 @@ public class ArrowsAndKeyboardNavigationHandler : IKeyboardNavigationHandler
         //todo: check XTFocus https://github.com/jinek/Consolonia/issues/105#issuecomment-2089015880
         private IInputRoot _owner;
 
-        public ArrowsAndKeyboardNavigationHandler(IKeyboardNavigationHandler keyboardNavigationHandler)
+        public ArrowsAndKeyboardNavigationHandler()
         {
-            _keyboardNavigationHandler = keyboardNavigationHandler;
+            _keyboardNavigationHandler = new KeyboardNavigationHandler();
```

Comment:
Could you please explain why is it null? May be in new version we need to register the handler somewhere? I mean may be in older version it was registered by something else which is not in the game anymore?

Reply:
The old code base used to instantiate a KeyboardHandler in the ConsoloniaApplication and then pass it in to our registered instance as a singleton. So both the keyboard handler and the ArrowsAndKeyboard handler were singletons.

The new Avalonia code base has KeyboardHandler as a transient in the AvaloniaLocator, which means it is created on demand. According to documentation and discussions AvaloniaLocator is an internal resource resolver and not meant for general purpose dependency resolution.  We are attempting to replace the one that is registered with the one that we have which wraps it. Soo...I looked at the registration of the dependency in the Avalonia side of things and it's just newing up a new instance, making it transient.  I was not able to create a new registration in AvaloniaLocator for the outer navigator without creating a recursive dependency, so I elected to just instantiate it in the class.

If you have a cleaner way of doing that I'm open to it.